journal/src/main/scala/com/evolutiongaming/kafka/journal/Journals.scala

```diff
@@ -268,15 +268,28 @@ object Journals {
               eventRecord    <- read(head, stream)
             } yield eventRecord
 
-            stream.stateful(from) { case (seqNr, a) =>
-              if (seqNr <= a.seqNr) {
-                val seqNr1 = a
-                  .seqNr
-                  .next[Option]
-                (seqNr1, Stream[F].single(a))
-              } else {
-                (seqNr.some, Stream[F].empty)
-              }
+            val empty = Option.empty[EventRecord[A]]
+
+            stream.map(Option(_)).concat(Stream.single(empty)).stateful(empty) {
+              case (None, None) =>
+                None -> Stream.empty
+
+              case (None, Some(record)) =>
+                record.some.some -> Stream.empty
+
+              case (Some(prevRecord), Some(record)) if prevRecord.seqNr == record.seqNr =>
+                val logError = log.warn(s"Duplicated events found during recovery: $prevRecord and $record").toStream
```

Comment:
it should be part of `debug` log, I don't like idea that because of database problems we generate such logs…